examples/minwebgl/mapgen_tiles_rendering/src/main.rs

```diff
@@ -0,0 +1,305 @@
+    let _ = body.append_child(&image);
+    image.set_id(&format!("{path}"));
+    let _ = image.style().set_property("visibility", "hidden");
+    let _ = image.style().set_property("position", "absolute");
+    let _ = image.style().set_property("top", "0");
+    let _ = image.style().set_property("width", "10px");
+    let _ = image.style().set_property("height", "10px");
+    image.set_cross_origin(Some("anonymous"));
+    let img = image.clone();
+    let on_load_callback: Closure<dyn Fn()> = Closure::new(move || on_load_callback(&img));
+    image.set_onload(Some(on_load_callback.as_ref().unchecked_ref()));
+    on_load_callback.forget();
+    let origin = window.location().origin().expect("Should have an origin");
+    let url = format!("{origin}/static/{path}");
+    image.set_src(&url);
+    image
+}
+
+fn init() {
+    gl::browser::setup(Default::default());
+
+    let window = web_sys::window().expect("Should have a window");
+    let document = window.document().expect("Should have a document");
+    let body_style = document.body().unwrap().style();
+    let _ = body_style.set_property("margin", "0");
+
+    set_resize_callback();
+    set_load_callback();
+}
+
+fn prepare_vertex_attributes() {
+    let gl = gl::context::retrieve_or_make().unwrap();
+
+    let position_data: [f32; 12] = [
+        -1., -1., -1., 1., 1., 1., 
+        -1., -1., 1., -1., 1., 1.
+    ];
+
+    let uv_data: [f32; 12] = [
+        0., 1., 0., 0., 1., 0., 
+        0., 1., 1., 1., 1., 0.
+    ];
+
+    let position_slot = 0;
+    let position_buffer = gl::buffer::create(&gl).unwrap();
+    gl::buffer::upload(&gl, &position_buffer, &position_data, GL::STATIC_DRAW);
+
+    let uv_slot = 1;
+    let uv_buffer = gl::buffer::create(&gl).unwrap();
+    gl::buffer::upload(&gl, &uv_buffer, &uv_data, GL::STATIC_DRAW);
+
+    let vao = gl::vao::create(&gl).unwrap();
+    gl.bind_vertex_array(Some(&vao));
+    gl::BufferDescriptor::new::<[f32; 2]>()
+        .stride(2)
+        .offset(0)
+        .attribute_pointer(&gl, position_slot, &position_buffer)
+        .unwrap();
+    gl::BufferDescriptor::new::<[f32; 2]>()
+        .stride(2)
+        .offset(0)
+        .attribute_pointer(&gl, uv_slot, &uv_buffer)
+        .unwrap();
+    gl.bind_vertex_array(None);
+    gl.bind_vertex_array(Some(&vao));
+}
+
+fn create_mvp() -> ndarray_cg::Mat<4, 4, f32, DescriptorOrderColumnMajor> {
+    let gl = gl::context::retrieve_or_make().unwrap();
+
+    let width = gl.drawing_buffer_width() as f32;
+    let height = gl.drawing_buffer_height() as f32;
+    let aspect_ratio = width / height;
+
+    let perspective_matrix =
+        ndarray_cg::d2::mat3x3h::perspective_rh_gl(70.0f32.to_radians(), aspect_ratio, 0.1, 1000.0);
+
+    let t = (0.0, 0.0, 0.0);
+    let translate = ndarray_cg::Mat::< 4, 4, _, DescriptorOrderColumnMajor >::_fill(0.0).raw_set
+    ([
+      1.0, 0.0, 0.0, t.0,
+      0.0, 1.0, 0.0, t.1,
+      0.0, 0.0, 1.0, t.2,
+      0.0, 0.0, 0.0, 1.0,
+    ]);
```

Comment:
Merge master and refactor your code using updated math types like F32x4x4. F32x4x4::from_row_major/from_column_major are preferred when you want to  create a matrix